python/pyarrow/tests/test_pandas.py

```diff
@@ -3531,11 +3533,11 @@ def test_dictionary_from_pandas_specified_type():
     assert result.type.equals(typ)
     assert result.to_pylist() == ['a', 'b']
 
-    # mismatching values type -> raise error (for now a deprecation warning)
+    # mismatching values type -> raise error
     typ = pa.dictionary(index_type=pa.int8(), value_type=pa.int64())
-    with pytest.warns(FutureWarning):
+    with pytest.raises(pa.ArrowInvalid):
         result = pa.array(cat, type=typ)
-    assert result.to_pylist() == ['a', 'b']
+    assert result.to_pylist() == ["a", "b"]
```

Comment:
this line can be removed now? (it's actually testing again the `result` of the previous case above)

Reply:
removed